### DISCUSSION TITLE
Additional Filtering

### DISCUSSION THREAD

**Discussion by TimSBenjamin** [philosowaffle/peloton-to-garmin#193] [Ideas]
I was going to start investigating additional filtering and contributing it but I wonder if this has been tried or not done for a particular reason before I start.

The scenario I'm thinking of is that sometimes I ride the Peloton bike, sometimes I ride my own bike using a Peloton workout. In the later scenario I often create duplicate workouts. I'm avoiding this right now by prompting for # of workouts to download. I was thinking of adding filter logic to look at total distance which could filter out workouts not actually done on the peloton bike. I could also like at something like app vs. bike as well but I assume it would be similar logic just a different field.

Has this been attempted before, any learnings before I start?

**Comment by philosowaffle** [comment]
This is not something that has been requested or tried before.  We've got a sample cycling payload [available here](https://github.com/philosowaffle/peloton-to-garmin/blob/master/UnitTests/Data/cycling_workout.json#L2123) and you can also turn on the `Json` flag in the [Format Config](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#format-config) to generate a similar file locally for your app workout and compare to see what fields make sense to use.

Whatever field is used, it will likely need to be evaluated on a per Workout Type basis. For instance, I may want to capture my Meditation or Yoga Flows, which don't have any distance metrics, but want to avoid converting the duplicate bike workout from the app.
